src/components/e-input.vue

```diff
@@ -296,39 +310,42 @@
          * @event enter
          */
         this.$emit('enter');
-        this.$parent.$emit('enter');
       },
 
       /**
        * Calculates the width of the slot content and sets it as a padding-right to the input-field.
        */
       setSlotSpacings() {
-        if (this.$refs.slot) {
-          const slotWidth = this.$refs.slot.clientWidth;
+        if (this.slot) {
+          const slotWidth = this.slot.clientWidth;
 
-          this.$refs.input.style.paddingRight = `${slotWidth + 10}px`;
+          if (this.input) {
+            this.input.style.paddingRight = `${slotWidth + 10}px`;
+          }
         }
       },
 
       /**
        * Selects the value of the input field.
        */
       selectValue() {
-        if (this.$props.value) {
+        if (this.value) {
           // Needed to select a number value on Chrome.
-          this.$refs.input.select();
+          this.input?.select();
 
           // Timeout is needed that it works on all browsers (without there are problems on Safari, Edge, iOS)
           if ('ontouchstart' in window) {
             setTimeout(() => {
-              this.$refs.input.setSelectionRange(0, this.$props.value.length);
+              const selectionRange = typeof this.value === 'string' ? this.value.length : this.value;
```

Comment:
We should make the value a String by default. `<input>` will only return strings anyway.

Reply:
mmh what if the input type is number?

Reply:
@janwidmer to my knowledge even a field of type number will return a string value. Can you please check? If this is true we wouldn't return a number in `onInput` either. But I see that it would be more convenient in some cases, but then again it might better be handled on the parent (e.g. `add-to-cart` component).

Reply:
adjust to return type string